x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -6,98 +6,100 @@
  */
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import createContainer from 'constate';
-import { useCallback, useReducer } from 'react';
+import { useCallback } from 'react';
 import { buildEsQuery, Filter, Query, TimeRange } from '@kbn/es-query';
-import DateMath from '@kbn/datemath';
 import type { SavedQuery } from '@kbn/data-plugin/public';
+import { debounce } from 'lodash';
 import type { InfraClientStartDeps } from '../../../../types';
 import { useMetricsDataViewContext } from './use_data_view';
-import { useKibanaTimefilterTime } from '../../../../hooks/use_kibana_timefilter_time';
-
-const DEFAULT_FROM_MINUTES_VALUE = 15;
+import { useSyncKibanaTimeFilterTime } from '../../../../hooks/use_kibana_timefilter_time';
+import { useHostsUrlState, INITIAL_DATE_RANGE, DEFAULT_QUERY } from './use_hosts_url_state';
 
 export const useUnifiedSearch = () => {
-  const [, forceUpdate] = useReducer((x: number) => x + 1, 0);
-
+  const { state, dispatch, getRangeInTimestamp, getTime } = useHostsUrlState();
   const { metricsDataView } = useMetricsDataViewContext();
   const { services } = useKibana<InfraClientStartDeps>();
   const {
     data: { query: queryManager },
   } = services;
 
-  const [getTime, setTime] = useKibanaTimefilterTime({
-    from: `now-${DEFAULT_FROM_MINUTES_VALUE}m`,
-    to: 'now',
+  useSyncKibanaTimeFilterTime(INITIAL_DATE_RANGE, {
+    from: state.dateRange.from,
+    to: state.dateRange.to,
   });
-  const { queryString, filterManager } = queryManager;
-
-  const currentDate = new Date();
-  const fromTS =
-    DateMath.parse(getTime().from)?.valueOf() ??
-    new Date(currentDate.getMinutes() - DEFAULT_FROM_MINUTES_VALUE).getTime();
-  const toTS = DateMath.parse(getTime().to)?.valueOf() ?? currentDate.getTime();
 
-  const currentTimeRange = {
-    from: fromTS,
-    to: toTS,
-  };
+  const { filterManager } = queryManager;
 
-  const submitFilterChange = useCallback(
+  const onSubmit = useCallback(
     (query?: Query, dateRange?: TimeRange, filters?: Filter[]) => {
-      if (filters) {
-        filterManager.setFilters(filters);
+      if (query || dateRange || filters) {
+        const newDateRange = dateRange ?? getTime();
+
+        if (filters) {
+          filterManager.setFilters(filters);
+        }
+        dispatch({
+          type: 'setQuery',
+          payload: {
+            query: query ? query : DEFAULT_QUERY,
```

Comment:
When we only set a filter, this function is called passing `query` as `undefined`, but it could be that in the hook state, `query` does have a previously set value. If we do this, it will reset whatever value `query` holds in the state. It's best to let the [reducer](https://github.com/elastic/kibana/pull/144181/files#diff-46e56847f0a8bf5427cc4c1c93d280065ee9256862d4f553179dec3e8ddd7d61R57) take care of the `query` and `filter` state, in case either of them is undefined.

Reply:
That makes sense, thank you! We don't need that logic here as the reducer handles that case, fixed 👍